### ISSUE TITLE
docker_container succeeds with empty result

### ISSUE THREAD

**Issue by mrkurt** [ansible/ansible#20543]
(Copying from https://github.com/ansible/ansible-modules-core/issues/5558 since I don't see it in this repo)

##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

docker_container

##### ANSIBLE VERSION

```
ansible 2.2.1.0
```

##### SUMMARY
The single ansible_docker_container key is being scrubbed from the ansible_facts result value, due to the fix for CVE-2016-3096 from issue ansible/ansible#15925. This is due to the return value's fact key having a prefix used for docker connection vars (ansible_docker_).

We probably don't want to add exceptions to the CVE fix, so it seems that the only reasonable fix is a breaking change to docker_container's return value (renaming the fact to docker_container or something else, or not returning as a fact.

##### STEPS TO REPRODUCE
Use the docker_container module in any way on affected versions and look at the output- it returns an empty ansible_facts dict instead of the actual module results.

##### EXPECTED RESULTS
Module return value as documented (including the ansible_docker_container subdict and data).

##### ACTUAL RESULTS
Empty ansible_facts dictionary.

**Comment by tgagor** [comment]
I could confirm that behaviour (just wanted to raise issue but found this one). Old `docker` module work like expected, but `docker_container` do not return facts. This started after last ansible update (from Ubuntu ppa), it was working earlier.

**Comment by sivel** [comment]
This appears to have been introduced in a6fff939

Because `docker_container` is returning a fact called `ansible_docker_container` it is removed as being a restricted key:

```
 [WARNING]: Removed restricted key from module data: ansible_docker_container
```

**Comment by tgagor** [comment]
Fingers crossed :)

**Comment by sivel** [comment]
We'll have to figure out what to do here.  In it's current form it is of course broken.

My preferred solution would be to remove setting `ansible_facts` and require playbook authors to `register` the results.

The other option is to change the key name.

This will have to be discussed a bit more among the core team.

**Comment by mrkurt** [comment]
Registering results would work for my case.

**Comment by tgagor** [comment]
I checked code diff you provided and it doesn't look like simple rename will do the trick.

**Comment by jctanner** [comment]
cc @chouseknecht

**Comment by bcoca** [comment]
closing as dupe of #22323 which was fixed via #22324
